lib/AST/Expr.cpp

```diff
@@ -1231,6 +1231,9 @@ SourceRange TupleExpr::getSourceRange() const {
   } else {
     // Scan forward for the first valid source loc.
     for (Expr *expr : getElements()) {
+      // Skip default arguments - they inherit the location of their parent.
+      if (isa<DefaultArgumentExpr>(expr))
```

Comment:
I see that in `coerceCallArguments` when we create `DefaultArgumentExpr` we give it `arg->getStartLoc()` which is supposed to be a location of "argument list". I think `DefaultArgumentExpr` should implement `getStartLoc` and return `{}` from it to indicate that it has to source location. This way we wouldn't need to touch TupleExpr, WDYT?

Reply:
I'm not even sure why would it implement `getSourceRange` with a single location, maybe we should just remove `Loc` from `DefaultArgumentExpr`?

Reply:
Yeah, makes sense – thanks!